src/accelerate/utils/transformer_engine.py

```diff
@@ -82,3 +82,41 @@ def has_transformer_engine_layers(model):
         if isinstance(m, (te.LayerNorm, te.Linear, te.TransformerLayer)):
             return True
     return False
+
+
+def contextual_fp8_autocast(fn, fp8_recipe, use_during_eval=False):
```

Comment:
Why "contextual" in the name?

Reply:
Originally it was going to be based on if the model is in `.eval()`, but I decided to switch it and give user choice.